schema/divisions/defs.yaml

```diff
@@ -89,6 +89,26 @@ description: Common schema definitions for divisions theme
           type: string
           minLength: 1
           pattern: ^(\S.*)?\S$    # Leading and trailing whitespace are not allowed.
+    perspectives:
+      description: Collection of perspectives of single feature.
+      type: object
+      unevaluatedProperties: false
+      required: [holders, includes]
+      properties:
+        holders:
+          description: Array of perspective holders.
+          type: array
+          items: { "$ref": "#/$defs/typeDefinitions/perspective" }
+          minItems: 1
+          uniqueItems: true
+        includes:
+          description:
+            Indicates if all counteries from the collection
+            or all except those in collection acknowledge the feature as is.
+
+            If value is true, only holders within collection see entity as it is represented.
+            If value is false, everyone except those in collection see entity as it is represented.
```

Comment:
On reflection, I think this isn't going to work as written. If we think this change is very important to pursue, we will need to think harder about the model and/or the documentation of what it means.

A superficial issue with the description is that it refers to *countries*, whereas the model is meant to be forward-compatible with adding new entity types, for example supranational entities like the UN or other blocs of countries, or maybe entities that don't have a recognized ISO-3166-1 alpha-2 country code but view themselves as being countries.

If you replace *countries* with *perspective holders* it seems to fix the superficial issue, but then we hit the conceptual issue.

The description says:

> If value is false, everyone except those in collection see entity as it is represented.

But who is *everyone*?

If we believe that only countries will ever have perspectives, that works. *Everyone* is *all countries* and then we can reason with "all countries except the ones listed".

But if other things can have perspectives, like supranational entities, then who is *everyone*? Does *everyone* include the UN? If yes, then does just saying "exclude RU" also imply "exclude UN"?

----

My conclusion is that the positive include model is forward-compatible with different holder types, but the include/exclude model is not.

So I think we have to ask ourselves:

1. How likely it is that we will need other types of perspective holders other than countries that have country codes.
2. How important is the exclusion mechanism? Is it a nice-to-have or a must-have?
3. Is there a third way to model this that we haven't thought of yet?

Reply:
1. If I understanding this question is referring to `Indicates if all countries from the collection`, I think we can change this to different wording or we can leave it as is, and update once we add new types of "holders" if ever.
2. Must-have, assume we have 2 Argentina entities, 
   i. One that does not contain part of Antartica, this one will have "Exclude AR"
   i. And another one that will contain part of Antartica that will have "Include AR"
I can't see how we could do it without exclude.
3. Not that I know of.

Reply:
> Must-have, assume we have 2 Argentina entities,
> One that does not contain part of Antartica, this one will have "Exclude AR"
> And another one that will contain part of Antartica that will have "Include AR"
> I can't see how we could do it without exclude.

You would have one Argentina that is from Argentina's perspective ("AR"), and one Argentina that is from everyone else's perspective (all other country codes).

Reply:
PR as updated, can this conversation be resolved?